kadi/commands/states.py

```diff
@@ -103,6 +102,32 @@ def disable_grating_move_duration():
     MechMove.apply_move_duration = apply_move_duration
 
 
+class Transition(dict):
+    """
+    Dict of transitions at a given date.
+
+    This is a dict of {state_key: state_val} where state_val is either a value
+    or a TransitionCallback object.
+    """
+
+    state_constraints = None
+
+    def __init__(self, *args, **kwargs):
+        if args and isinstance(args[0], str):
+            date = args[0]
+            args = args[1:]
+        else:
+            date = None
```

Comment:
I reviewed coverage, and this isn't tested.  That's probably OK for this use case.

Reply:
That actually suggests nicer code with a class variable `date = None` and then `self.date = args[0]` above. I'll try that (not 100% sure if setting `self.date` before `super().__init__()` will work, but easy to test).

Reply:
I did just notice my mac pytests are skipping all of test_commands_create_archive_regress (I guess I don't have enough MP_DIR) which is a big deal so that possible coverage wasn't 100% complete.